qemu/tests/win_heavyload.py

```diff
@@ -139,9 +139,18 @@ def add_option(cmd, key, val):
             if not installed and download_url:
                 utils.system("rm -f %s/HeavyLoad*.exe" % tmp_dir)
                 session.cmd("del /f /s %sHeavyLoad*.exe" % dst)
-            if loop_session_cmd(session, check_running_cmd):
-                if not loop_session_cmd(session, stop_cmd):
-                    raise error.TestFail("Unable to terminate heavyload "
-                                         "process")
+            i = 0
+            while i < 3:
+                if session.cmd_status(check_running_cmd) == 0:
+                    try:
+                        session.cmd(stop_cmd, timeout=120)
+                        break
+                    except:
+                        i += 1
+                        continue
+                else:
+                    break
+            else:
+                raise
```

Comment:
I think it would be better to use 'for' instead of 'while' here.
for i in range(3):
...
else:
...